src/providers/dart_commands.ts

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+import * as as from "../analysis/analysis_server_types";
+import * as editors from "../editors";
+import * as vs from "vscode";
+import { Analyzer } from "../analysis/analyzer";
+
+export class DartCommands  {
+	private analyzer: Analyzer;
+	constructor(context: vs.ExtensionContext, analyzer: Analyzer) {
+		this.analyzer = analyzer;
+
+        context.subscriptions.push(
+            vs.commands.registerTextEditorCommand("dart.organizeDirectives", this.organizeDirectives, this)
+        );
+	}
+
+    organizeDirectives(editor: vs.TextEditor, editBuilder: vs.TextEditorEdit) {
+        if (!editors.hasActiveDartEditor()) {
+            vs.window.showWarningMessage("No active Dart editor.");
+            return;
+        }
+
+        this.analyzer.editOrganizeDirectives({ file: editor.document.fileName }).then((response) => {
+            let edit: as.SourceFileEdit = response.edit;
+			if (edit.edits.length == 0) {
+                vs.window.showInformationMessage("No changes from organize directives.");
```

Comment:
It's nice to let the user know that the operation did run, so they don't think it didn't take or the key binding wasn't correct. The toast vscode uses is persistent though, so the user would have to dismiss it each time. I'll remove the notification here and we can see how it feels.